tests/_bootstrap.php

```diff
@@ -40,3 +42,44 @@
 // By default, do not enable the Custom Tables v1 implementation in tests.
 putenv( 'TEC_CUSTOM_TABLES_V1_DISABLED=1' );
 $_ENV['TEC_CUSTOM_TABLES_V1_DISABLED'] = 1;
+
+/**
+ * Fake Order Instance's lock mechanism without actually using DB transactions to avoid leaking in tests.
+ *
+ * @return void
+ */
+function tests_et_enable_fake_lock_mechanism() {
+	$lock_key = 'post_content_filtered';
+	uopz_set_return( Order::class, 'lock_order', function ( int $order_id ) use ( $lock_key ) {
+		tribe( Order::class )->generate_lock_id();
+
+		return (bool) DB::query(
+			DB::prepare(
+				"UPDATE %i set $lock_key = %s where ID = $order_id and $lock_key = ''",
+				DB::prefix( 'posts' ),
+				tribe( Order::class )->get_lock_id()
+			)
+		);
+	}, true );
+
+	uopz_set_return( Order::class, 'unlock_order', function ( int $order_id ) use ( $lock_key ) {
+		return (bool) DB::query(
+			DB::prepare(
+				"UPDATE %i set $lock_key = '' where ID = $order_id",
+				DB::prefix( 'posts' )
+			)
+		);
+	}, true );
+}
+
+/**
+ * Disable the fake lock mechanism for Order instances.
+ *
+ * @return void
+ */
+function tests_et_disable_fake_lock_mechanism() {
```

Comment:
This seems too complicated.
The lock mechanism is based on setting the value of the `post_content_filtered` to the lock key.
To simulate, in tests, a locked order, you can just set the value of `post_content_filtered` to a value that is not empty.
To simulate an unlocked order, you can set the value to an empty string.
Why not use this simpler approach?
Mocking the Order `lock_order` and `unlock_order` methods does not actually cover them.